warn/scrapers/la.py

```diff
@@ -0,0 +1,394 @@
+import logging
+import os
+import re
+from datetime import datetime
+from pathlib import Path
+
+import pdfplumber
+from bs4 import BeautifulSoup
+
+from .. import utils
+from ..cache import Cache
+
+__authors__ = ["chriszs"]
+__tags__ = ["html", "pdf"]
+
+logger = logging.getLogger(__name__)
+
+
+def scrape(
+    data_dir: Path = utils.WARN_DATA_DIR,
+    cache_dir: Path = utils.WARN_CACHE_DIR,
+) -> Path:
+    """
+    Scrape data from Lousiana.
+
+    Keyword arguments:
+    data_dir -- the Path were the result will be saved (default WARN_DATA_DIR)
+    cache_dir -- the Path where results can be cached (default WARN_CACHE_DIR)
+
+    Returns: the Path where the file is written
+    """
+    # Fire up the cache
+    cache = Cache(cache_dir)
+
+    # The basic configuration for the scrape
+    state_code = "la"
+    base_url = "https://www.laworks.net/"
+    file_base = "Downloads_WFD"
+
+    # Download the root page
+    url = f"{base_url}Downloads/{file_base}.asp"
+    html = utils.get_url(url).text
+
+    # Save it to the cache
+    cache_key = f"{state_code}/{file_base}.html"
+    cache.write(cache_key, html)
+
+    # Parse out the links to WARN notice PDFs
+    links = _parse_links(html)
```

Comment:
If this function is just a couple lines of code and is only getting called once, is it necessary? Could we do it inline?

Reply:
We could. There are maybe two competing code clarity demands:
1. Implementation hiding (do you really need to know the details of how I'm parsing the links at a high level? ideally, we just call it by a name and use that)
2. Complexity reduction (do you really need to make me go seek out a function definition to find out how this works? ideally, it's all simple enough we can do it inline)

Add to that the possibility of refactoring the definition of how we parse links without having to change the top level flow.

I'm opting for more of 1 in this case, because this got more complicated than most other scrapers, and I wanted to keep the top level flow simple, but I did course correct during this refactor and get rid of a couple one-line functions that were doing something simple, because they obscured more than they enlightened. Could do the same here if you think that's wisest.

Reply:
If you feel strongly about the approach, I'm fine with keeping it. My lizard brain was living more in the DRY world where you only need a function if you are going to reuse it.

Reply:
I killed this function.